src/libstd/sys/unix/os.rs

```diff
@@ -102,14 +94,22 @@ pub fn error_string(errno: i32) -> String {
 }
 
 pub fn getcwd() -> io::Result<PathBuf> {
-    let mut buf = [0 as c_char; BUF_BYTES];
-    unsafe {
-        if libc::getcwd(buf.as_mut_ptr(), buf.len() as libc::size_t).is_null() {
-            Err(io::Error::last_os_error())
-        } else {
-            Ok(bytes2path(CStr::from_ptr(buf.as_ptr()).to_bytes()))
+    super::fill_bytes_buf(|mut buf| {
+        unsafe {
+            let ptr = buf.as_mut_ptr() as *mut libc::c_char;
+            Ok(if !libc::getcwd(ptr, buf.capacity() as libc::size_t).is_null() {
+                let len = CStr::from_ptr(buf.as_ptr() as *const libc::c_char).to_bytes().len();
+                buf.set_len(len);
+                Ok(PathBuf::from(OsString::from_bytes(buf).unwrap()))
+            } else {
+                let error = io::Error::last_os_error();
+                if error.raw_os_error().unwrap() == libc::ERANGE {
```

Comment:
While `unwrap()` will always succeed, it's not idiomatic, so let's compare against `Some(libc::ERANGE)`